internal/test/consensus/scenarios/intersect_origin_one_rollforward/configurator.sh

```
@@ -0,0 +1,113 @@
+# the pool's keys + configs under /configs/1/ for the cardano-node
+# service to mount read-only.
+#
+# Inherits the cardano-foundation testnet-generation-tool baked into
+# the shared internal/test/consensus/Dockerfile.configurator image;
+# this script is mounted in at runtime.
+
+set -euo pipefail
+
+# Write-then-rename helper so we never observe a partial config file.
+write_file() {
+    local tmp_file="${1}_$(tr </dev/urandom -dc A-Za-z0-9 | head -c16)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)